### ISSUE TITLE
Optional chaining operator bug

### ISSUE THREAD

**Issue by jonatanvm** [microsoft/TypeScript#48174]
# Bug Report

The optional chaining operator has some inconsistent behaviour as shown in the playground link.

### 🔎 Search Terms

optional chaining operator 

### 🕗 Version & Regression Information

- This raises an unexpected error (Object is possibly 'undefined')
- Have tested this in nightly, v4.6.2 and v4.4.4
- This is the behavior in every version I tried, and I reviewed the FAQ for entries about Type System Behavior

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhgLhtATgSzAcwNoF0YA+MYArgDZkwC8MeAUGgGYwAUcA-AHRkCmWUACxgA+GAEYAlDADedGPJihIIXtxCYWkugF86dJdBgAjarXxwIrVBhz4ipChIbMWRrr35DRkmXIUGVHjUNLV19cENgJGssPEJickoaOIsrKHRYuwTHZ1Zgdz5MQRFxKVkFRQjA4M0nbSA)

### 💻 Code

```ts
const a: string[] | null = []
if (a?.length > 1) {
    console.log(1)
}

const b = [] as (string[] | null)
if (b?.length > 1) {
    console.log(1)
}

const c: string[] | null = [] as (string[] | null)
if (c?.length > 1) {
    console.log(1)
}
```

### 🙁 Actual behavior

(b) and (c) give errors (Object is possibly 'undefined').

### 🙂 Expected behavior

(a), (b) and (c) should all work, without errors.

**Comment by MartinJohns** [comment]
This is not a bug, but working as intended.

The optional chaining operator is irrelevant here. You can't compare a `undefined` value using the >` operator. Here's a simplified example of your issue:

```typescript
declare const a: number | undefined
if (a > 0) { }
```

See #17801.

You can use the coalesce operator and make your code more explicit: `?? 0 > 1`
